### ISSUE TITLE
dbg! put spaces around ::

### ISSUE THREAD

**Issue by pickfire** [rust-lang/rust#81209]
While trying out `Ipv4Addr` for https://github.com/rust-lang/rust/pull/81202

I tried this code:

https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=d4e00191c64a2608f3306178bff743c8

```rust
use std::net::Ipv4Addr;

fn main() {
    dbg!(format!("{:#?}", Ipv4Addr::new(127, 0, 0, 1)));
}
```

I expected to see this happen:
```
[src/main.rs:4] format!("{:#?}", Ipv4Addr::new(127, 0, 0, 1)) = "127.0.0.1"
```

Instead, this happened:
```
[src/main.rs:4] format!("{:#?}", Ipv4Addr :: new(127, 0, 0, 1)) = "127.0.0.1"
```

### Meta

`rustc --version --verbose`: Same happens for nightly
```
rustc 1.49.0 (e1884a8e3 2020-12-29)
binary: rustc
commit-hash: e1884a8e3c3e813aada8254edfa120e85bf5ffca
commit-date: 2020-12-29
host: x86_64-unknown-linux-gnu
release: 1.49.0
```

**Comment by jonas-schievink** [comment]
This is just what `stringify!` does, I believe

**Comment by pickfire** [comment]
But why shouldn't it be normalized without spaces around `::`?

**Comment by clarfonthey** [comment]
I didn't realise that, and it's a good point.

It could potentially be worthwhile to make `stringify!` format the code in a reasonably nice way if it's valid Rust syntax, but that could induce a lot of overhead. Really depends on what people think is best and makes the most sense.

**Comment by GrigorenkoPV** [comment]
Pretty sure it got fixed in 1.76

https://godbolt.org/z/3bnrvE5M7

**Comment by clarfonthey** [comment]
Seems like this should be closed, then.
